channels/tee-channel-js/src/TeeChannel.ts

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import dynamicProto from "@microsoft/dynamicproto-js";
+import { IChannelControlsAI, IConfig } from "@microsoft/applicationinsights-common";
+import {
+    BaseTelemetryPlugin, IAppInsightsCore, IChannelControls, IConfigDefaults, IConfiguration, IPlugin, IProcessTelemetryContext,
+    IProcessTelemetryUnloadContext, IProcessTelemetryUpdateContext, ITelemetryItem, ITelemetryPluginChain, ITelemetryUnloadState,
+    ITelemetryUpdateState, createProcessTelemetryContext, initializePlugins, onConfigChange, proxyFunctions
+} from "@microsoft/applicationinsights-core-js";
+import { arrForEach, isArray, objDeepFreeze, objFreeze, throwError } from "@nevware21/ts-utils";
+import { ChannelControllerPriority, IChannelController, _IInternalChannels, createChannelControllerPlugin } from "./ChannelController";
+import { ITeeChannelConfig } from "./Interfaces/ITeeChannelConfig";
+
+const ChannelValidationMessage = "Channel has invalid priority - ";
+
+const defaultTeeChannelConfig: IConfigDefaults<ITeeChannelConfig> = objDeepFreeze({
+});
```

Comment:
I need to add the "defaults" in here (even if they are undefined) so that they become dynamic.